src/lib/common/differenceSignal.cpp

```diff
@@ -40,6 +54,41 @@ bool DifferenceSignal::encodeDifferenceSignalToBuffer(double *d_n, size_t &numSa
   return rval;
 }
 
+/*!
+ * @brief Clears the signal buffer
+ */
 void DifferenceSignal::clear() { r_i = 0; }
 
+/*!
+ * @brief Checks if the signal buffer is full
+ * @return True if the buffer is full, false otherwise
+ */
 bool DifferenceSignal::isFull() { return r_i == r_n; }
+
+/*!
+ * @brief Calculates the mean of the signal
+ * @return The mean of the signal
+ */
+double DifferenceSignal::signalMean() {
+    if(!r_i) {
+        return 0.0;
+    }
+    double sum = 0.0;
+    for (uint32_t i = 0; i < r_i; i++) {
+        sum += r[i];
+    }
+    return sum / r_i;
+}
+
+/*!
+ * @brief Gets the reference signal
+ * @param r0[out] The reference signal
+ * @return True if the reference signal was retrieved, false otherwise
+ */
+bool DifferenceSignal::getReferenceSignal(double &r0) {
```

Comment:
@russelldeguzman this is the residual required for decoding, right?

Reply:
Yes thats right!